### DISCUSSION TITLE
querystring an nested arrays

### DISCUSSION THREAD

**Discussion by kevinruscoe-popcreative** [livewire/livewire#4577] [7. Bugs (archived)]
Hello, I make use of an array to filter my models. I have a array named `$filters` which looks like this

```php
$filters = [
  'price' => [],
  'colour' => []
];
```

When I update my UI the array get appended to and I use Laravel to filter. This works great, however, when I use this in conjunction with the `$querystring` property, it seems to overwrite the array upon reloading values from the querystring. If I have just 'price' filters active, it removes the 'colour' key from the array.

Is there a proper way around this? At the moment I've got it working by re-merging the arrays before the querystring values are injected like:

```php
public function boot() {
  $this->_filters = $this->filters;
}

public function mount() {
  $this->filters = array_merge($this->_filters, $this->filters);
}
```

**Comment by Coyote6** [comment]
That is pretty much the exact same way I solved the issue. I ran from boot() and booted() though to keep my mount() clean for classes that extend my base component.  The issue I am running into with an array is trying to figure out a way to tell it to have an exception for empty strings.

```php
public $filters = [
	'search' => '',
	'id' => '',
	'name' => '',
	'client' => ''
];
protected $queryString = [
	'filters' => ['except' => ''] 
];
```

Guessing it doesn't keep it from showing because keys are set.

**Reply by Coyote6** [reply]
Updated code to support a pseudo query string... Instead of creating a new property and then merging the arrays, you can remove the query strings and set the values yourself.

```php
public $filters = [
	'search' => '',
	'id' => '',
	'name' => '',
	'client' => ''
];

// protected $queryString = [
//	'filters' => ['except' => ''] 
//];

public function booted () {
	collect (request()->input())->filter (function ($value, $filter) {
		if ($filter == 'filters') {	
        		if (is_array ($value)) {
			    	foreach ($value as $k => $v) {
				    	$this->filters[$k] = $v;
			    	}
			}
		}
    	});
}
```
